File: Carthage/Checkouts/SwiftTheme/OCDemo/SwitchNightCell.m
//
//  SwitchNightCell.m
//  SwiftTheme
//
//  Created by Gesen on 16/5/27.
//  Copyright © 2016年 Gesen. All rights reserved.
//

#import "SwitchNightCell.h"
#import "Global.h"
#import "MyThemes.h"
#import "OCDemo-Swift.h"

@implementation SwitchNightCell

- (void)awakeFromNib {
    [super awakeFromNib];
    
    self.title.theme_textColor = globalTextColorPicker;
    self.nightIcon.theme_image = [ThemeImagePicker pickerWithNames:@[@"icon_night_dark", @"icon_night_dark", @"icon_night_dark", @"icon_night_light"]];
    
    [self updateNightSwitch];
    
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(updateNightSwitch) name:@"ThemeUpdateNotification" object:nil];
}

- (IBAction)changeNight:(UISwitch *)sender {
    [MyThemes switchNight:sender.on];
}

- (void)updateNightSwitch {
    self.nightSwitch.on = [MyThemes isNight];
}

@end

File: Carthage/Checkouts/SwiftTheme/Demo/SwitchNightCell.swift
//
//  SwitchNightCell.swift
//  Demo
//
//  Created by Gesen on 16/3/2.
//  Copyright © 2016年 Gesen. All rights reserved.
//

import UIKit

class SwitchNightCell: BaseCell {
    
    @IBOutlet weak var title: UILabel!
    @IBOutlet weak var nightIcon: UIImageView!
    @IBOutlet weak var nightSwitch: UISwitch!

    override func awakeFromNib() {
        super.awakeFromNib()

        title.theme_textColor = globalTextColorPicker
        nightIcon.theme_image = ["icon_night_dark", "icon_night_dark", "icon_night_dark", "icon_night_light"]
        
        updateNightSwitch()
        
        NotificationCenter.default.addObserver(self, selector: #selector(updateNightSwitch), name: NSNotification.Name(rawValue: ThemeUpdateNotification), object: nil)
    }
    
    @IBAction func changeNight(_ sender: UISwitch) {
        MyThemes.switchNight(isToNight: sender.isOn)
    }
    
    @objc private func updateNightSwitch() {
        nightSwitch.isOn = MyThemes.isNight()
    }

}
